mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -43,21 +40,146 @@ impl Settings {
         config.try_into()
     }
 
-    pub fn nested_bridge(&self) -> &NestedBridgeSettings {
-        &self.nested_bridge
+    pub fn upstream(&self) -> Option<&ConnectionSettings> {
+        self.upstream.as_ref()
+    }
+
+    pub fn remotes(&self) -> &Vec<ConnectionSettings> {
+        &self.remotes
     }
 
-    pub fn subscriptions(&self) -> &Subscriptions {
+    pub fn messages(&self) -> &MessagesSettings {
+        &self.messages
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for Settings {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Debug, serde_derive::Deserialize)]
+        struct Inner {
+            #[serde(flatten)]
+            nested_bridge: CredentialProviderSettings,
+
+            upstream: UpstreamSettings,
+
+            remotes: Vec<ConnectionSettings>,
+
+            messages: MessagesSettings,
+        }
+
+        let Inner {
+            nested_bridge,
+            upstream,
```

Comment:
Should these be separate if they are getting combined into one settings object?

Reply:
After looking at this more I agree. They are separate at first because it is easier to input, then get combined for ease of use in the application.

Reply:
this is used because the settings from env have a different format, but then they are combined in same struct